File: src/services/weatherService.js
import { WEATHER_DATABASE } from '../data/weatherData';

export const weatherService = {
  getWeatherByDistrict: async (districtName) => {
    // Phase 2: return axios.get(`/api/v1/weather?district=${districtName}`);
    return new Promise((resolve) => {
      setTimeout(() => {
        const data = WEATHER_DATABASE[districtName] || {
          district: districtName || "Ludhiana",
          state: "Punjab",
          temp: 30,
          feelsLike: 33,
          condition: "Partly Cloudy with Good Breeze",
          icon: "⛅",
          humidity: 68,
          windSpeed: "12 km/h N",
          rainProbability: 35,
          uvIndex: 6,
          soilMoisture: "Adequate (64%)",
          alerts: [
            {
              id: 'w-alert-gen',
              severity: "info",
              title: "🌾 Regular Farm Operations Favorable",
              message: "Weather is conducive for weeding, gentle irrigation, and routine crop scouting."
            }
          ],
          forecast7Days: [
            { day: "Today", tempMax: 31, tempMin: 23, condition: "Partly Cloudy", rain: 35, icon: "⛅" },
            { day: "Fri", tempMax: 32, tempMin: 24, condition: "Passing Rain", rain: 45, icon: "🌦️" },
            { day: "Sat", tempMax: 30, tempMin: 22, condition: "Thunderstorms", rain: 65, icon: "⛈️" },
            { day: "Sun", tempMax: 33, tempMin: 24, condition: "Sunny", rain: 15, icon: "☀️" },
            { day: "Mon", tempMax: 34, tempMin: 25, condition: "Clear", rain: 10, icon: "☀️" },
            { day: "Tue", tempMax: 35, tempMin: 26, condition: "Warm", rain: 5, icon: "☀️" },
            { day: "Wed", tempMax: 33, tempMin: 24, condition: "Cloudy", rain: 25, icon: "⛅" }
          ],
          farmingAdvice: [
            "Schedule irrigation in early morning to minimize evapotranspiration losses.",
            "Scout crop undersides for early pest presence.",
            "Maintain clean bunds around the field."
          ]
        };
        resolve(data);
      }, 300);
    });
  }
};

File: backend/src/main/java/com/agrisahay/service/WeatherService.java
package com.agrisahay.service;

import org.springframework.stereotype.Service;

import java.util.*;

@Service
public class WeatherService {

    public Map<String, Object> getWeatherByDistrict(String district) {
        String distName = (district == null || district.isBlank()) ? "Karur" : district;

        Map<String, Object> data = new HashMap<>();
        data.put("district", distName);
        data.put("temp", 33);
        data.put("condition", "Tropical Warm & Sunny");
        data.put("humidity", 68);
        data.put("windSpeedKmH", 14);
        data.put("uvIndex", "8 (High)");
        data.put("icon", "☀️");

        List<Map<String, Object>> forecast = new ArrayList<>();
        String[] days = {"Today", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"};
        int[] temps = {33, 34, 32, 30, 31, 33, 34};
        int[] rains = {15, 20, 65, 80, 40, 10, 5};

        for (int i = 0; i < days.length; i++) {
            Map<String, Object> day = new HashMap<>();
            day.put("day", days[i]);
            day.put("tempMax", temps[i]);
            day.put("rainChancePercent", rains[i]);
            day.put("icon", rains[i] > 50 ? "🌧️" : "☀️");
            forecast.add(day);
        }

        data.put("forecast7Days", forecast);
        data.put("advisory", "Moderate humidity in " + distName + ". Delay heavy paddy nitrogen top dressing if rain occurs on Wed.");

        return data;
    }
}
